design/types_test.go

```diff
@@ -109,3 +109,148 @@ func TestIsPrimitive(t *testing.T) {
 		}
 	}
 }
+
+func TestPrimitiveIsCompatible(t *testing.T) {
+	var (
+		b    bool    = true
+		i    int     = 1
+		i8   int8    = 2
+		i16  int16   = 3
+		i32  int32   = 4
+		ui   uint    = 5
+		ui8  uint8   = 6
+		ui16 uint16  = 7
+		ui32 uint32  = 8
+		i64  int64   = 9
+		ui64 uint64  = 10
+		f32  float32 = 10.1
+		f64  float64 = 20.2
+		s    string  = "string"
+		bs   []byte  = []byte("bytes")
+	)
+	cases := map[string]struct {
+		p        Primitive
+		values   []interface{}
+		expected bool
+	}{
+		//			values:   []interface{}{b, i, i8, i16, i32, ui, ui8, ui16, ui32, i64, ui64, f32, f64, s, bs},
```

Comment:
Oops, it's my temporary comment. https://github.com/goadesign/goa/pull/1579/commits/f094c7660955f602c7dc34ba563b1fe49b25af3f